Makefile

```diff
@@ -263,6 +263,15 @@ tests/pyret/main2.jarr: phaseA tests/pyret/main2.arr  $(TEST_FILES)
 pyret-test: phaseA tests/pyret/main2.jarr
 	$(NODE) tests/pyret/main2.jarr
 
+tests/io-tests/main.jarr: phaseA tests/io-tests/main.jarr  $(TEST_FILES)
+	$(TEST_BUILD) \
+		--outfile tests/io-tests/main.jarr \
+		--build-runnable tests/io-tests/main.arr
+
+.PHONY : pyret-io-test
+pyret-io-test: phaseA tests/io-tests/main.jarr
+	echo "hello" | $(NODE) tests/io-tests/main.jarr
```

Comment:
can definitely do better here. Now that I've moved away from file descriptor, an option is `mock-stdin` which should work.